wikimedia/wikipedia-ios
# Update Notifications Center bar button item in Settings view

**Phabricator:** N/A

### Notes
Updates the Notifications Center bar button item in the Settings view to present/hide as needed. Additionally, this PR updates the bell button item image to be either the badged or unbadged variant depending on the user's unread notifications count.

The requirements in order to see the bell bar button item in the Settings view are:
- The user has disabled the Explore feed (meaning the Settings view is presented as a tab, not modally over the Explore feed)
- The user is authenticated/logged in

For all modal presentations of the Settings view, the Close button should be the only navigation bar button item. For all remaining presentations of the Settings view where the user is not logged in, no Notifications Center or Close button should appear in the navigation bar.

There are a few patterns in this PR that I don't feel are very direct and could be improved, but I generally followed the existing patterns for the sake of conformity.


### Test Steps
Confirm:
- User is logged in, presenting Settings modally, close button is shown
- User is logged out, presenting Settings modally, close button is shown
- User is logged in, presenting Settings as tab, notifications button is shown
- User is logged out, presenting Settings as tab, notifications button is not shown

Also confirm bell badge image itself:
- User is logged in, presenting Settings as tab, they have unread notifications, notifications image with badge is shown
- User is logged in, presenting Settings as tab, they have no unread notifications, notifications image without badge is shown
